### PR TITLE
Implemented a preliminary LivingDeathEvent

### PR DESCRIPTION
**Pull request by DarkArc**
It's not complete, but it at least makes living death events fire in Sponge.

### CODE REVIEW

File: src/main/java/org/spongepowered/mod/event/SpongeModEventManager.java [hunk 1]
```diff
@@ -94,6 +95,7 @@
                     .put(BlockEvent.class, net.minecraftforge.event.world.BlockEvent.class)
                     .put(BlockUpdateEvent.class, net.minecraftforge.event.world.BlockEvent.NeighborNotifyEvent.class)
                     .put(LivingEvent.class, net.minecraftforge.event.entity.living.LivingEvent.class)
+                    // .put(LivingDeathEvent.class, net.minecraftforge.event.entity.living.LivingDeathEvent.class)
```
Comment (Aaron1011) [inline src/main/java/org/spongepowered/mod/event/SpongeModEventManager.java]: Why are you commenting this out?

File: src/main/java/org/spongepowered/mod/mixin/core/event/entity/living/MixinEventLivingDeath.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        return getEntity().getLocation();
+    }
+
+    /*
+    private static net.minecraftforge.event.entity.living.LivingDeathEvent fromSpongeEvent(org.spongepowered.api.event.entity.living.LivingDeathEvent spongeEvent) {
+        LivingDeathEvent event = new LivingDeathEvent((EntityLivingBase) spongeEvent.getEntity(), Dam);
+        ((IMixinEvent) event).setSpongeEvent(spongeEvent);
+        return event;
+    }
+    */
```
Comment (DarkArc) [inline src/main/java/org/spongepowered/mod/mixin/core/event/entity/living/MixinEventLivingDeath.java]: Because I didn't finish this function yet...

**Comment by DarkArc** [comment]
@Aaron1011 any suggestions on how to actually implement fromSpongeEvent, seeing as how the sponge event doesn't concern itself with damage causes, and the constructor for the forge version of the event requires a damage cause?

**Comment by ryantheleach** [comment]
I'd change the API for the Sponge event to get the damage result/cause that resulted from the killing blow. Even though the death event shouldn't be cancellable in sponge, it would be useful to know what was the final cause.

**Comment by DarkArc** [comment]
@ryantheleach right, but that's why the cause API exist. Though, I don't know what it's state is.

**Comment by gabizou** [comment]
The cause API is also going through a change with SpongePowered/SpongeAPI#712 as well, but I agree, the causes is the correct place to store the direct cause for the event. If anything, the `DamageCause` can be recreated or vice versa.

**Comment by Zidane** [comment]
Events API has changed drastically.

**Pull request closed without merging**